src/components/Login/LoginOTPCard.tsx

```diff
@@ -35,20 +40,23 @@ interface LoginOTPCard {
   mobileNumber: string;
   codeKey: string;
   endpoint: string;
+  lastResendWarningMessageRef: MutableRefObject<string>;
```

Comment:
slightly confused with the use of a `MutableRefObject` here for the warning message, can you elaborate a bit on this?

Reply:
the `MutableRefObject` remembers the data stored in the warning message through rerenders. does it make the code a lot less readable? i can change it to be a variable outside the `LoginContainer` component if so i think

Reply:
it seems slightly complex for the warning message, whats the data stored in the warning message?

Reply:
it's just a string 😅

Reply:
hmm, this is to be able to pass the message into the individual card ah?

Reply:
yeahh

Reply:
hmm, is there a reason we have to keep the last message state actually? is it due to some checks haha

Reply:
ahh, is it for the alert before resend?

Reply:
yup this is for the case where the user moves on to the `LoginOTPCard` and the message was triggered by an entry in the `LoginMobileNumberCard`

Reply:
would it be better for it to be stored just as variable outside `LoginContainer`? and also, the warning messages will always be for the case of requesting too many OTPs?

Reply:
the warning message will warn the user just before he requests the last OTP before he/she times out. the message is as follows
> After 1 more OTP request, you will have to wait 3 minutes before trying again.

> would it be better for it to be stored just as variable outside LoginContainer? 
sure, i can try that

Reply:
hmm i tried it and it didn't work. i was following [this article](https://www.emgoto.com/storing-values-with-useref/) when i suggested that. i guess it's not that easy to change parent state with child components

Reply:
ahhh, i see i see. Hmm, maybe let's leave this for now? We can create a tech chore to remind ourselves to look into this to see if there's a better way haha @sebastianquek what do you think ahah

Reply:
also @ddwsd i think for this we can add the new AlertModal we are using!!